### DISCUSSION TITLE
When using uploadListingImage, "alt_text" can be set when updating using "listing_image_id", but isn't set with initial upload using "image"

### DISCUSSION THREAD

**Discussion by entseed** [etsy/open-api#1182] [Q&A]
I am only able to set the "alt_text" when updating an existing image.  When I try to set it during an initial image upload, the alt_text is empty.
I assume this is a bug.

I use etsy-php-sdk.
The following code will correctly upload an image and set the correct rank, but the alt text will not be set:
`
		$listing_image = $etsy_listing->uploadImage(["image" => $image_path, "rank"=>$ii, "alt_text"=>"MY ALT TEXT"]]);
`

To remedy this, I make an additional call to set the alt_text:
`
		$listing_image = $etsy_listing->uploadImage(["listing_image_id" => $listing_image->toArray()["listing_image_id"], "alt_text"=>"MY ALT TEXT"]);
`

I don't know whether, how, or if I should submit this as a bug, but I thought the work around might be useful for someone who's trying to include alt_text.

**Comment by showtroylove** [comment]
I reported this issue when this forum first opened.  No reply was forthcoming.  I will test whether this work-around works using other languages / tech.

**Comment by showtroylove** [comment]
> I reported this issue when this forum first opened. No reply was forthcoming. I will test whether this work-around works using other languages / tech.

It seems that this technique has problems. I assigned specific alt text to each image, but it appears to be applied incorrectly. While it does add the alt text, it seems to retain the value from a previous image upload. Additionally, the second upload triggers an error when called using the image ID.

Since all image slots are filled, to avoid the error, I must call the second upload with the 'overwrite' set to true. However, this causes peculiar behavior in the system because I'm effectively referring to an image I'm also asking it to overwrite (the same image). I will continue testing this image upload and alt text API to understand the issue better.

**Comment by entseed** [comment]
When I do the uploadImage the second time, I'm not actually uploading the image again (I'm not overwriting the picture).  All I'm doing is just passing the "listing_image_id" and "alt_text" parameters and it seems to work.

Since there doesn't seem to be a way to store alt data for etsy image, one of the reasons I use the alt_text field is to keep track of what etsy image is there (I set it to $filename plus a crc32 checksum of the $file) so that I can automatically sync my woocommerce and etsy images.  Etsy won't let you have 0 images, so I had to to make sure I worked around that in my code.

**Reply by showtroylove** [reply]
Yes, I understand that, but when you perform your second call it is in fact performing an upload.  It not doing so with an image you provide during he call, but instead fetching it from it's cloud storage (your implied image storage for your account).   

I also wanted you to confirm if it's save the results as expected.  In other words, can you confirm your alt tags are the matched correctly to the image?
